### PR TITLE
Controller refactoring

### PR DESCRIPTION
**Pull request by david-mears-dfe**
- Rewrote a method to read more analogously to its neighbours
- Removed method that needlessly reimplements Devise functionality

### CODE REVIEW

File: app/controllers/jobseekers/passwords_controller.rb [hunk 1]
```diff
@@ -14,9 +14,4 @@ def ensure_reset_password_period_valid
   def after_sending_reset_password_instructions_path_for(_resource)
     jobseekers_check_your_email_password_path
   end
-
-  def after_resetting_password_path_for(resource)
```
Comment (cesidio) [inline app/controllers/jobseekers/passwords_controller.rb:18]: This is needed to sign in the and redirect the user to `jobseekers_saved_jobs_path`

Comment (david-mears-dfe) [reply app/controllers/jobseekers/passwords_controller.rb:18]: See the commit message! It's not needed.

Comment (david-mears-dfe) [reply app/controllers/jobseekers/passwords_controller.rb:18]: Maybe when you were writing this controller, the routes.rb did not yet contain 'jobseeker_root', which is what Devise uses to determine the path for after sign in and for after resetting the password.

**Review by cesidio — APPROVED** [review]

File: app/controllers/jobseekers/passwords_controller.rb (hunk 1 above)
Comment (cpjmcquillan) [reply app/controllers/jobseekers/passwords_controller.rb:18]: We need to sign in though still, don't we?

**Pull request merged**